### PR TITLE
Fix part of #5404: Migrate away from onBackPressed 

### PR DESCRIPTION
**Pull request by dattasneha**
## Explanation

Fixes part of #5404 

This PR migrates deprecated `onBackPressed` usage to `OnBackPressedDispatcher` callback in the following activities and presenters.

- AdministratorControlsActivity
- AppVersionActivity
- ProfileAndDeviceIdActivity
- MarkChaptersCompletedActivity
- MarkStoriesCompletedActivity
- MarkTopicsCompletedActivity
- ReadingTextSizeActivityPresenter
- ExplorationActivityPresenter
- ResumeLessonActivityPresenter

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by dattasneha** [comment]
Hi,@adhiamboperes PTAL.

File: app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivity.kt [hunk 1]
```diff
@@ -121,7 +128,7 @@ class AdministratorControlsActivity :
     }
   }
 
-  override fun onBackPressed() {
+  private fun handleTheOnBackPressed() {
```
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivity.kt]:
```suggestion
  private fun handleBackPress() {
```

File: app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivity.kt [hunk 2]
```diff
@@ -134,8 +141,7 @@ class AdministratorControlsActivity :
     if (fragment is ProfileEditFragment) {
       administratorControlsActivityPresenter.handleOnBackPressed()
     } else {
-      @Suppress("DEPRECATION") // TODO(#5404): Migrate to a back pressed dispatcher.
-      super.onBackPressed()
+      finish()
```
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivity.kt:147]: When you finish the current activity, does it go back to the previous expected activity? (including the other files in the PR with the same implementation)

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @dattasneha! I have a couple of comments, PTAL.

File: app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivity.kt (hunk 1 above)
Comment (dattasneha) [reply app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivity.kt]: Done, updated the function name.

File: app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivity.kt (hunk 2 above)
Comment (dattasneha) [reply app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivity.kt:147]: Yes, I have run and tested all the files in the PR to verify if, when the current activity is finished, it correctly returns to the previous expected activity, except for `ReadingTextSizeActivity`  in the options package as I could not find the Activity in the app's Options navigation menu.

**Comment by dattasneha** [comment]
@adhiamboperes PTAL. Also, should I work on the remaining activities as well, since the other PR seems to be inactive?

**Comment by adhiamboperes** [comment]
> @adhiamboperes PTAL. Also, should I work on the remaining activities as well, since the other PR seems to be inactive?

Thanks, once this PR is merged, please feel free to create a new one with the remaining activities.

**Comment by adhiamboperes** [comment]
@dattasneha PTAl at failing lint check.

**Comment by dattasneha** [comment]
@adhiamboperes PTAL.

**Comment by adhiamboperes** [comment]
Thanks @dattasneha. The changes look good, but will take another look once CI passes.

**Comment by dattasneha** [comment]
> > @adhiamboperes PTAL. Also, should I work on the remaining activities as well, since the other PR seems to be inactive?
> 
> Thanks, once this PR is merged, please feel free to create a new one with the remaining activities.

Thanks. I have created a new PR for the remaining activities.

**Review by adhiamboperes — APPROVED** [review]
LGTM, thanks @dattasneha!

**Comment by dattasneha** [comment]
Hi, @adhiamboperes, is there anything else needed to be done by me on this PR ?

**Pull request merged**